### DISCUSSION TITLE
Alter the value in custom validation function?

### DISCUSSION THREAD

**Discussion by ItalyPaleAle** [go-playground/validator#957] [Q&A] [answered]
Is there a way to alter the value in a custom validation function?

I need to write validators that do not just accept or reject a value, but alter it, sanitizing the value or modifying it in other ways (for example, I need to "normalize" the Unicode form of certain strings).

I tried things like:

```go
d.validate.RegisterValidationCtx("myvalidator", func(ctx context.Context, fl validator.FieldLevel) bool {
	fl.Field().SetString("newvalue")
	return true
}, false)
```

But this is causing a panic as, it seems, `fl` is not assignable.

**Comment by deankarn** [comment, marked as answer]
No this is jot possible using validator for multiple reasons, please see/search for the multiple closed issues asking the same :)

For sanitization, prior to validation, this library exists https://github.com/go-playground/mold

**Reply by ItalyPaleAle** [reply]
Thanks for the quick response!

I tried searching for "normalize", "modify", "alter" (and variations) but couldn't find anything :) mold seems to be helpful, I'll take it for a spin.
